File: delivery-frame-spec/tests/run_all.py
#!/usr/bin/env python3
"""One-command delivery-family self-check. Run before shipping the four delivery-* skills.

Runs, in order:
  1. test_family_integrity.py          — references resolve, atomic set present, version majors
  2. ../../delivery-execute-verify/tests/test_template_anchor_consistency.py
                                       — machine anchors + limited synonym lock
  3. run_fixture_tests.py              — 12 pass/neg handoff fixtures (hard profile)
  4. test_single_chain.py              — Standard 4-stage chain + cascade + High slice

Exit 0 only when every suite passes.
"""

from __future__ import annotations

import subprocess
import sys
from pathlib import Path

TESTS_DIR = Path(__file__).resolve().parent
SUITES = (
    TESTS_DIR / "test_family_integrity.py",
    TESTS_DIR.parent.parent / "delivery-execute-verify" / "tests" / "test_template_anchor_consistency.py",
    TESTS_DIR / "run_fixture_tests.py",
    TESTS_DIR / "test_single_chain.py",
)


def main() -> int:
    if hasattr(sys.stdout, "reconfigure"):
        sys.stdout.reconfigure(encoding="utf-8")
    failed = []
    for suite in SUITES:
        print(f"\n=== {suite.name} ===", flush=True)
        result = subprocess.run([sys.executable, str(suite)])
        if result.returncode != 0:
            failed.append(suite.name)
    if failed:
        print(f"\nSELF-CHECK FAILED: {', '.join(failed)}", file=sys.stderr)
        return 1
    print(f"\nSELF-CHECK PASSED: all {len(SUITES)} suites green — family is shippable")
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: delivery-frame-spec/tests/run_all.mjs
#!/usr/bin/env node
// @ts-check
/**
 * One-command delivery-family self-check. Run before shipping the four delivery-* skills.
 *
 * Runs, in order:
 *   1. test_family_integrity.mjs         — references resolve, atomic set present, version majors
 *   2. ../../delivery-execute-verify/tests/test_template_anchor_consistency.mjs
 *                                        — machine anchors + limited synonym lock
 *   3. run_fixture_tests.mjs             — 12 pass/neg handoff fixtures (hard profile)
 *   4. test_single_chain.mjs             — Standard 4-stage chain + cascade + High slice
 *
 * Exit 0 only when every suite passes. Zero dependencies (Node >= 18).
 */

import { spawnSync } from "node:child_process";
import { basename, dirname, join, resolve } from "node:path";
import { fileURLToPath } from "node:url";
import process from "node:process";

const TESTS_DIR = dirname(fileURLToPath(import.meta.url));
const SUITES = [
  join(TESTS_DIR, "test_family_integrity.mjs"),
  resolve(TESTS_DIR, "..", "..", "delivery-execute-verify", "tests", "test_template_anchor_consistency.mjs"),
  join(TESTS_DIR, "run_fixture_tests.mjs"),
  join(TESTS_DIR, "test_single_chain.mjs"),
];

function main() {
  /** @type {string[]} */
  const failed = [];
  for (const suite of SUITES) {
    console.log(`\n=== ${basename(suite)} ===`);
    const result = spawnSync(process.execPath, [suite], { stdio: "inherit" });
    if (result.status !== 0) failed.push(basename(suite));
  }
  if (failed.length) {
    console.error(`\nSELF-CHECK FAILED: ${failed.join(", ")}`);
    return 1;
  }
  console.log(`\nSELF-CHECK PASSED: all ${SUITES.length} suites green — family is shippable`);
  return 0;
}

process.exit(main());
